File: client/lib/suite_detail_etudiant/Model/CVmodel.dart
class CVmodel{
  List<String>? nomCompetance;

  List<String>? nomFormation;
  List<String>? etablissementF;
  List<String>? villeF;
  List<DateTime>? datedF;
  List<DateTime>? datefF;
  List<String>? DescriptionF;

  List<String>? nomSociete;
  List<DateTime>? datedS;
  List<DateTime>? datefS;
  List<String>? DescriptionS;

  List<String>? nomLangue;
  List<String>? NiveauLangue;

  List<String>? centreInteret;

  CVmodel({this.nomCompetance,this.nomFormation,
  this.etablissementF,this.villeF,this.datedF,this.datefF,this.DescriptionF,
    this.nomLangue,this.NiveauLangue,this.centreInteret,this.nomSociete,this.DescriptionS,this.datedS,this.datefS});

  Map<String,dynamic> toJson(){
    final Map<String,dynamic> comp =new Map<String,dynamic>();

    comp['nomCompetance']= nomCompetance;

    comp['nomFormation']= nomFormation;
    comp['etablissementF']= etablissementF;
    comp['villeF']= villeF;
    comp['datedF']= datedF;
    comp['datefF']= datefF;
    comp['DescriptionF']= DescriptionF;

    comp['nomSociete']= nomSociete;
    comp['nomSociete']= datedS;
    comp['nomSociete']= datefS;
    comp['nomSociete']= DescriptionS;

    comp['nomLangue']= nomLangue;
    comp['NiveauLangue']= NiveauLangue;

    comp['centreInteret']= centreInteret;
    return comp;
}

}

File: serveur2/models/cv.model.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const Cv = Schema(

  {
    username: {
      type: String,
      required: true,
      unique: true,
    },
    nom: { type: String, default: null },
    email: { type: String, default: null},
    Prenom: { type: String, default: null },
    Numerotel: { type: Number, default: null },
    Adresse: { type: String, default: null },
    Codepostale: { type: String, default: null },
    Ville: { type: String, default: null },
    Profil: {type: String, default: null },
    Realisation: {type: String, default: null },
  
    Competence :[{ nomCompetence:{ type: String, default: null } }],
    
    Formation :[
      { 
       nomFormation: { type: String, default: null},
       etablissementF:{ type: String, default: null },
       villeF:{ type: String, default: null },
       datedF:{ type: Date, default: null },
       datefF:{ type: Date, default: null },
       DescriptionF:{ type: String, default: null } 
       }],
  
    Stage :[{ nomSociete: { type: String, default: null }, 
      datedS:{ type: Date, default: null }, datefS:{ type: Date, default: null },
      DescriptionS:{ type: String, default: null }  }],
  
    CI :[{ nomCI: { type: String, default: null } }],
  
    langue :[{ nomLangue: { type: String, default: null }, NiveauLangue: { type: String, default: null } }],
  
    img:{type: String,default:""}

  },
  {
    timestamp: true,
  }
);

module.exports = mongoose.model("Cv", Cv);
